### ISSUE TITLE
PUT /v2/apps/{appId} to non-leader Marathon does not work

### ISSUE THREAD

**Issue by zcox** [mesosphere/marathon#2546]
We have 3 instances of Marathon 0.10.1 running in HA. We use `PUT /v2/apps/{appId}` to deploy changes to applications, and have noticed that if we send the HTTP PUT to the leader Marathon it works fine, but if we send the HTTP PUT to a non-leader Marathon, the HTTP request just gets back an empty response, and no actions are taken.

The non-leader Marathon logs show this:

```
2015-11-02_21:28:33.15515 [2015-11-02 21:28:33,154] INFO Proxying request to PUT http://mesos-master2.redacted.com:8080/v2/apps/my-app from mesos-master1.redacted.com:8080 (mesosphere.marathon.api.JavaUrlConnectionRequestForwarder$:288)
2015-11-02_21:28:43.20710 [2015-11-02 21:28:43,205] WARN /v2/apps/my-app (org.eclipse.jetty.servlet.ServletHandler:563)
2015-11-02_21:28:43.20714 java.lang.RuntimeException: while proxying
2015-11-02_21:28:43.20716   at mesosphere.marathon.api.LeaderProxyFilter.doFilter(LeaderProxyFilter.scala:147)
2015-11-02_21:28:43.20716   at com.google.inject.servlet.FilterDefinition.doFilter(FilterDefinition.java:163)
2015-11-02_21:28:43.20717   at com.google.inject.servlet.FilterChainInvocation.doFilter(FilterChainInvocation.java:58)
2015-11-02_21:28:43.20722   at com.google.inject.servlet.ManagedFilterPipeline.dispatch(ManagedFilterPipeline.java:118)
2015-11-02_21:28:43.20722   at com.google.inject.servlet.GuiceFilter.doFilter(GuiceFilter.java:113)
2015-11-02_21:28:43.20723   at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1467)
2015-11-02_21:28:43.20724   at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:501)
2015-11-02_21:28:43.20725   at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1086)
2015-11-02_21:28:43.20726   at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:429)
2015-11-02_21:28:43.20727   at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1020)
2015-11-02_21:28:43.20728   at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:135)
2015-11-02_21:28:43.20730   at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:116)
2015-11-02_21:28:43.20731   at com.codahale.metrics.jetty8.InstrumentedHandler.handle(InstrumentedHandler.java:192)
2015-11-02_21:28:43.20732   at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:154)
2015-11-02_21:28:43.20733   at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:116)
2015-11-02_21:28:43.20734   at org.eclipse.jetty.server.Server.handle(Server.java:370)
2015-11-02_21:28:43.20735   at org.eclipse.jetty.server.AbstractHttpConnection.handleRequest(AbstractHttpConnection.java:494)
2015-11-02_21:28:43.20736   at org.eclipse.jetty.server.AbstractHttpConnection.headerComplete(AbstractHttpConnection.java:971)
2015-11-02_21:28:43.20736   at org.eclipse.jetty.server.AbstractHttpConnection$RequestHandler.headerComplete(AbstractHttpConnection.java:1033)
2015-11-02_21:28:43.20737   at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:651)
2015-11-02_21:28:43.20738   at org.eclipse.jetty.http.HttpParser.parseAvailable(HttpParser.java:235)
2015-11-02_21:28:43.20739   at org.eclipse.jetty.server.AsyncHttpConnection.handle(AsyncHttpConnection.java:82)
2015-11-02_21:28:43.20740   at org.eclipse.jetty.io.nio.SelectChannelEndPoint.handle(SelectChannelEndPoint.java:696)
2015-11-02_21:28:43.20742   at org.eclipse.jetty.io.nio.SelectChannelEndPoint$1.run(SelectChannelEndPoint.java:53)
2015-11-02_21:28:43.20743   at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:608)
2015-11-02_21:28:43.20745   at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:543)
2015-11-02_21:28:43.20746   at java.lang.Thread.run(Thread.java:745)
2015-11-02_21:28:43.20747 Caused by: java.net.SocketTimeoutException: Read timed out
2015-11-02_21:28:43.20748   at java.net.SocketInputStream.socketRead0(Native Method)
2015-11-02_21:28:43.20748   at java.net.SocketInputStream.read(SocketInputStream.java:152)
2015-11-02_21:28:43.20749   at java.net.SocketInputStream.read(SocketInputStream.java:122)
2015-11-02_21:28:43.20750   at java.io.BufferedInputStream.fill(BufferedInputStream.java:235)
2015-11-02_21:28:43.20751   at java.io.BufferedInputStream.read1(BufferedInputStream.java:275)
2015-11-02_21:28:43.20752   at java.io.BufferedInputStream.read(BufferedInputStream.java:334)
2015-11-02_21:28:43.20753   at sun.net.www.http.HttpClient.parseHTTPHeader(HttpClient.java:687)
2015-11-02_21:28:43.20753   at sun.net.www.http.HttpClient.parseHTTP(HttpClient.java:633)
2015-11-02_21:28:43.20754   at sun.net.www.protocol.http.HttpURLConnection.expect100Continue(HttpURLConnection.java:1021)
2015-11-02_21:28:43.20756   at sun.net.www.protocol.http.HttpURLConnection.getOutputStream(HttpURLConnection.java:1106)
2015-11-02_21:28:43.20757   at mesosphere.marathon.api.JavaUrlConnectionRequestForwarder$$anonfun$copyRequestBodyToConnection$1$1.apply(LeaderProxyFilter.scala:246)
2015-11-02_21:28:43.20758   at mesosphere.marathon.api.JavaUrlConnectionRequestForwarder$$anonfun$copyRequestBodyToConnection$1$1.apply(LeaderProxyFilter.scala:245)
2015-11-02_21:28:43.20759   at mesosphere.marathon.io.IO$.using(IO.scala:89)
2015-11-02_21:28:43.20760   at mesosphere.marathon.api.JavaUrlConnectionRequestForwarder.copyRequestBodyToConnection$1(LeaderProxyFilter.scala:245)
2015-11-02_21:28:43.20761   at mesosphere.marathon.api.JavaUrlConnectionRequestForwarder.copyRequestToConnection$1(LeaderProxyFilter.scala:256)
2015-11-02_21:28:43.20761   at mesosphere.marathon.api.JavaUrlConnectionRequestForwarder.forward(LeaderProxyFilter.scala:298)
2015-11-02_21:28:43.20762   at mesosphere.marathon.DebugModule$MetricsBehavior$$anonfun$invoke$1.apply(DebugConf.scala:50)
2015-11-02_21:28:43.20763   at mesosphere.marathon.metrics.Metrics.timed(Metrics.scala:24)
2015-11-02_21:28:43.20764   at mesosphere.marathon.DebugModule$MetricsBehavior.invoke(DebugConf.scala:49)
2015-11-02_21:28:43.20765   at mesosphere.marathon.api.LeaderProxyFilter.doFilter(LeaderProxyFilter.scala:143)
2015-11-02_21:28:43.20767   ... 26 more
2015-11-02_21:28:43.20908 [2015-11-02 21:28:43,208] INFO 10.103.226.200 -  -  [02/Nov/2015:21:28:33 +0000] "PUT /v2/apps/my-app HTTP/1.1" 200 0 "-" "curl/7.43.0" (mesosphere.chaos.http.ChaosRequestLog$$EnhancerByGuice$$76a5a899:15)
```

**Comment by pierlo-upitup** [comment]
Possibly a duplicate of https://github.com/mesosphere/marathon/issues/2540

**Comment by zcox** [comment]
I'm not sure this is a dup of #2540 - we are not using HTTPS, and loading the UI from a non-leader Marathon works fine. We just are unable to make changes to applications via non-leaders.

**Comment by aquamatthias** [comment]
This is fixed in 0.11. Please update to latest version 0.11.1.
